internal/kratos/kratos.go

```diff
@@ -0,0 +1,69 @@
+// Copyright © 2023 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+package kratos
+
+import (
+	"context"
+	"net/url"
+
+	"go.opentelemetry.io/otel/attribute"
+
+	"github.com/ory/hydra/v2/driver/config"
+	"github.com/ory/hydra/v2/x"
+	client "github.com/ory/kratos-client-go"
+	"github.com/ory/x/otelx"
+)
+
+type (
+	dependencies interface {
+		config.Provider
+		x.HTTPClientProvider
+		x.TracingProvider
+	}
+	Provider interface {
+		Kratos() Client
+	}
+	Client interface {
+		DisableSession(ctx context.Context, kratosSessionID string) error
+	}
+	Default struct {
+		dependencies
+	}
+)
+
+func New(d dependencies) Client {
+	return &Default{dependencies: d}
+}
+
+func (k *Default) DisableSession(ctx context.Context, kratosSessionID string) (err error) {
+	ctx, span := k.Tracer(ctx).Tracer().Start(ctx, "kratos.DisableSession")
+	otelx.End(span, &err)
+
+	adminURL, ok := k.Config().KratosAdminURL(ctx)
+	if !ok {
+		span.SetAttributes(attribute.Bool("skipped", true))
+		span.SetAttributes(attribute.String("reason", "kratos admin url not set"))
+		return nil
+	}
+
+	if kratosSessionID == "" {
+		span.SetAttributes(attribute.Bool("skipped", true))
+		span.SetAttributes(attribute.String("reason", "kratos session ID is empty"))
+		return nil
+	}
+
+	kratos := client.NewAPIClient(k.clientConfiguration(ctx, adminURL))
+	_, err = kratos.IdentityApi.DisableSession(ctx, kratosSessionID).Execute()
+
+	return err
+
+}
+
+func (k *Default) clientConfiguration(ctx context.Context, adminURL *url.URL) *client.Configuration {
+	configuration := client.NewConfiguration()
+	configuration.Servers = client.ServerConfigurations{{URL: adminURL.String()}}
+	configuration.HTTPClient = k.HTTPClient(ctx).StandardClient()
```

Comment:
This might need
```suggestion
	configuration.HTTPClient = k.HTTPClient(ctx, httpx.ResilientClientWithTracer(k.Tracer(ctx).Tracer())).StandardClient()```

Reply:
No, it's good, because the registry is already injecting the tracer here:

https://github.com/ory/hydra/blob/fb0707373b29a699290315820756cfa98380b0b5/driver/registry_base.go#L362

Reply:
sounds good